### DISCUSSION TITLE
Show only the first word from the command as a tab name ?

### DISCUSSION THREAD

**Discussion by ghost** [kovidgoyal/kitty#5100] [Q&A]
I have shell integration enabled and when i type  `nvim .config/kitty/kitty.conf ` it shows as a tab name, is there a way to only show the first name out of this sentence (in this case show only nvim as the tab name )

**Comment by kovidgoyal** [comment]
If you want custom title handling, add no-title to shell_integration in
kitty.conf and write your own title setting function in your shell's rc
files.

**Reply by ghost** [reply]
I tried so, i found this script on the internet that works .
but i couldn't manage to make it only show the first word of a command  
```
function set-title-preexec() {
  printf "\e]2;%s\a" "$1"
}
autoload -Uz add-zsh-hook
add-zsh-hook preexec set-title-preexec
```

**Reply by kovidgoyal** [reply]
So find another script on the net that shows you how to extract the first word from a string in zsh and combine it with this one.
